modules/EmailAddresses/EmailAddress.php

```diff
@@ -122,6 +122,6 @@ public function confirmOptIn()
         global $timedate;
         $date = new DateTime();
         $this->confirmed_opt_in_date = $date->format($timedate::DB_DATETIME_FORMAT);
-        $this->confirm_opt_in = 1;
+        $this->confirm_opt_in = 'confirm-opt-in';
```

Comment:
Using `'confirm-opt-in'` string constant as enum value but many places in this PR using it as `'confirmed-opt-in'` probably  one of these is incorrect.

Reply:
I have updated it